src/process/terminalsource.rs

```diff
@@ -194,6 +194,26 @@ impl ColorableTerminal {
     }
 }
 
+impl TerminalInner {
```

Comment:
Nit: ordering in this module is a bit of a mess, but please keep impls below their type definitions.

Reply:
Was trying to be consistent with the file which seemed to be types before impls (mostly). But fine with changing this.

Reply:
Yes, I think that would be nice. I'll clean up the other items after this is merged.

(Note that CI is failing on a formatting issue.)